idaes/apps/grid_integration/multiperiod/design_and_operation_models.py

```diff
@@ -0,0 +1,138 @@
+        raise ValueError(
+            f"The attribute `{attr}` does not exist on the {obj.__class__.__name__} instance passed. Make sure that `{attr}` is defined for that object."
+        )
+
+
+@declare_process_block_class("DesignModel")
+class DesignModelData(SkeletonUnitModelData):
+    """
+    Class for containing design model ...
+    """
+
+    CONFIG = SkeletonUnitModelData.CONFIG()
+    CONFIG.declare(
+        "model_func",
+        ConfigValue(
+            doc="Function that builds the design model",
+        ),
+    )
+    CONFIG.declare(
+        "model_args",
+        ConfigValue(
+            default={},
+            doc="Dictionary containing arguments needed for model_func",
+        ),
+    )
+
+    # noinspection PyAttributeOutsideInit
+    def build(self):
+        super().build()
+
+        self.config.model_func(self, **self.config.model_args)
+
+
+@declare_process_block_class("OperationModel")
+class OperationModelData(SkeletonUnitModelData):
+    """
+    Class for containing design model ...
+    """
+
+    CONFIG = SkeletonUnitModelData.CONFIG()
+    CONFIG.declare(
+        "model_func",
+        ConfigValue(
+            doc="Function that builds the design model",
+        ),
+    )
+    CONFIG.declare(
+        "model_args",
+        ConfigValue(
+            default={},
+            doc="Dictionary containing arguments needed for model_func",
+        ),
+    )
+
+    CONFIG.declare(
+        "design_blk",
+        ConfigValue(
+            default={},
+            doc="Dictionary containing arguments needed for model_func",
+        ),
+    )
+    CONFIG.declare(
+        "declare_op_vars",
+        ConfigValue(
+            default=True,
+            domain=In([True, False]),
+            doc="Should op_mode, startup, shutdown vars be defined?",
+        ),
+    )
+    CONFIG.declare(
+        "append_lmp_data",
+        ConfigValue(
+            default=True,
+            domain=In([True, False]),
+            doc="Should LMP data automatically be appended to the model?",
+        ),
+    )
+
+    # noinspection PyAttributeOutsideInit
+    def build(self):
+        super().build()
+
+        # Build the model
+        if self.config.declare_op_vars:
+            self.op_mode = Var(
+                within=Binary,
+                doc="Binary var: 1 if the unit is operating, 0 otherwise",
+            )
+
+            self.startup = Var(
+                within=Binary,
+                doc="Binary var: 1 if the startup is initiated, 0 otherwise",
+            )
+
+            self.shutdown = Var(
+                within=Binary,
+                doc="Binary var: 1 if the shutdown is initiated, 0 otherwise",
+            )
+
+        if self.config.append_lmp_data:
+            self.LMP = Param(
+                initialize=1,
+                mutable=True,
+                doc="Parameter: Will be updated to LMP value at given time",
+            )
+
+        self.config.model_func(self, **self.config.model_args)
```

Comment:
@dlakes94 Need to pass `design_blk` as an argument if it is specified. Alternatively, we can delete lines 86-92 and let the user pass it in `model_args`.

Reply:
Pushed the responsibility to the user so design_blk is now expected in model_args if the user's model depends on the design_blk.